pkg/aws/client.go

```diff
@@ -967,9 +970,12 @@ func (c *awsClient) DetachRolePolicies(roleName string) error {
 		if err != nil {
 			return err
 		}
-		isTruncated = *resp.IsTruncated
+		isTruncated = resp.IsTruncated
 		marker = resp.Marker
-		attachedPolicies = append(attachedPolicies, resp.AttachedPolicies...)
+		for _, attachedPolicy := range resp.AttachedPolicies {
```

Comment:
If we create a slice of `iamtypes.AttachedPolicy` instead of a slice of pointers, we can remove the conversion here.

Reply:
Good spot thank you